packages/runtime/datastore/src/fluidHandle.ts

```diff
@@ -24,6 +21,22 @@ export class FluidObjectHandle<T extends FluidObject = IFluidObject> implements
         return this.routeContext.isAttached;
     }
 
+    /**
+     * Tells whether the object of this handle is visible in the container.
+     */
+    private get visible(): boolean {
+        /**
+         * If the object of this handle is attached, it is visible in the container. This is a work around the scenario
+         * where the object becomes visible but the handle does not get this notification. For example, handles to a DDS
+         * other than the default handle won't know if the DDS becomes visible after the handle was created.
+         */
+        if (this.isAttached) {
```

Comment:
In fact, the handle for root data stores will always its visibility state as not visible. This is because attachGraph() will never be called on it. For non-root data stores, attachGraph() will be called when its handle is stored in a DDS.

Things work today because the handles of DDSes of root-data stores will update its visibility state. This happens when bindChannel() is called for the DDS which ends up calling handle.attachGraph().